slacker.go

```diff
@@ -146,17 +173,20 @@ func (s *Slacker) handleMessage(event *slack.MessageEvent) {
 		s.handleHelp(event.Channel)
 		return
 	}
-
 	response := NewResponse(event.Channel, s.rtm)
+	ctx, cancel := context.WithTimeout(context.Background(), s.timeout)
+	defer cancel()
```

Comment:
I have been thinking about this and I feel that if we were to introduce a timeout to the command requests, we should also be handling the "select" case internally, otherwise we would be promising a timeout but put it on the developer to worry about implementing the logic in every command handler.

I am thinking we should have the timeout logic perhaps inside the "cmd.Execute"? But again, we wouldn't be able to "cancel" whatever logic the user wrote.

So this made me think more about the context. Perhaps, we allow the user to pass in their own "context" and if no context was provided, we default to context.Background. Then it's up to the user how to leverage the power of the context. Then they would have to implement their own timeout logic inside their handlers.